frame/support/src/traits/hooks.rs

```diff
@@ -80,13 +85,20 @@ pub trait OnIdle<BlockNumber> {
 }
 
 #[impl_for_tuples(30)]
-impl<BlockNumber: Clone> OnIdle<BlockNumber> for Tuple {
+impl<BlockNumber: 'static + Copy + AtLeast32BitUnsigned> OnIdle<BlockNumber> for Tuple {
+	for_tuples!( const ON_IDLE_FUNCTIONS : &'static [fn(BlockNumber, crate::weights::Weight) -> crate::weights::Weight] = &[ #(Tuple::on_idle),* ]; );
+
 	fn on_idle(n: BlockNumber, remaining_weight: crate::weights::Weight) -> crate::weights::Weight {
 		let mut weight = 0;
-		for_tuples!( #(
+		let len = <Self as OnIdle<BlockNumber>>::ON_IDLE_FUNCTIONS.len();
+		let start_index = n % (len as u32).into();
+		let start_index = start_index
+			.try_into()
+			.unwrap_or_else(|_| panic!("Can not convert BlockNumber to usize"));
```

Comment:
```suggestion
			.expect("`start_index % len` always fits into `usize`, because `len` can be in maximum `usize::MAX`; qed");
```

Reply:
Unfortunately it is not possible because ``<BlockNumber as TryInto<usize>>::Error` doesn't implement `std::fmt::Debug`` 
So will save the text that you have been provide but remain `unwrap_or_else`